### PR TITLE
mint NFT - show USD value

### LINKED ISSUE #7157: Listing NFT for sale: Showing approximate price in USD
## Example that is already working
Its possible to see the approximate price in USD if you are using the Teleport Bridge feature on KodaDot. If you open Teleport and decide to teleport your assets, you are able to see how much of the assets you are Teleporting. Like this: 
Screenshot 2023-09-05 at 12 46 44

## Suggestion - Listing NFT for Sale
It would be very cool to have this option (to see the price in USD) also when you list NFT for sale:

1) After you create a collection and decide to add NFT to the collection and directly list it for sale. I checked on rmrk2 and assethub we don't have it now.
Create - NFT:
Screenshot 2023-09-05 at 12 49 00

2) When you decide to list the NFt you own for sale:
Profile - Collected/Created NFTs - List for sale
Screenshot 2023-09-05 at 12 52 16

Both these options should show how much you are setting the price for approximately. To make it easier for people to decide what price should their NFT have. And to make it more user friendly.

### LINKED ISSUE #7936: Minting: Show all prices also in USD
Adding NFTs to collection: https://kodadot.xyz/create/nft
I wanna see all these KSM/DOT amounts also in USD. It's necessary for a simpler user experience. 

Screenshot 2023-11-03 at 12 50 13

Partially similar issue: #7157

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type
- [x] Feature

## Needs QA check

- @kodadot/qa-guild please review

## needs design review

 - @exezbcz  please review

## Context

- [x] Closes #7936
- [x] Closes #7157

* note that on #7157 the interafce for listing has changed, making the changes requested for listing an nft irrelvent, instead - price is shown in the listing modal

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)

## Screenshot 📸

- [x] My fix has changed UI

image
image

### <samp>🤖 Generated by Copilot at fa34e4d</samp>

Added a feature to display the usd value of the nft sale price, deposit, and balance in the create nft component. Created a new composable function `useUsdValue` to convert token prices to usd prices based on the fiat store. Exported the `FiatPrice` type from the `stores/fiat.ts` file.

### <samp>🤖 Generated by Copilot at fa34e4d</samp>

> _Oh, we're the coders of the sea, and we work with composable_
> _We use `useUsdValue` to show the price in dollar_
> _Heave away, me hearties, heave away with `fiatStore`_
> _We'll raise the sail of NFTs with `FiatPrice` and more_

### CODE REVIEW

**Comment by yangwao** [comment]
- #7992 - just saying we should look on this as currently sometimes coingecko can go down and then it's unusable in transfer for example

**Comment by daiagi** [comment]
@preschian @roiLeo
If I may have your review plz 😎

File: components/create/CreateNft.vue [hunk 1]
```diff
@@ -354,6 +370,23 @@ const deposit = computed(() =>
...
+  isBasilisk.value ? chainSymbol.value.toLowerCase() : undefined,
+)
+
+const calculateUsdValue = (amount) => {
+  // remove comma from amount - required becuase bsx balance is formatted string
+  const parsedAmount = parseFloat(amount?.replace(/,/g, '') || '0')
+  return toUsdPrice(parsedAmount, tokenType.value)
+}
+
+const SalePriceUsd = computed(() => toUsdPrice(form.salePrice, tokenType.value))
```
Comment (roiLeo) [inline components/create/CreateNft.vue:386]: **suggestion**: camelCase

File: components/create/CreateNft.vue [hunk 2]
```diff
@@ -192,13 +195,25 @@
       <div>
         <div class="is-flex has-text-weight-medium has-text-info">
           <div>{{ $t('mint.deposit') }}:&nbsp;</div>
-          <div data-testid="create-nft-deposit-amount">
-            {{ deposit }} {{ chainSymbol }}
+          <div>
+            <span data-testid="create-nft-deposit-amount-ksm">
```
Comment (roiLeo) [inline components/create/CreateNft.vue:199]: and what about `-dot`?

File: components/create/CreateNft.vue [hunk 3]
```diff
@@ -354,6 +370,23 @@ const deposit = computed(() =>
   (Number(totalItemDeposit.value) * form.copies).toFixed(4),
 )
 
+// usd value
+
+// when left undefined urlPrefix will be used
+const tokenType = computed(() =>
+  isBasilisk.value ? chainSymbol.value.toLowerCase() : undefined,
+)
```
Comment (roiLeo) [inline components/create/CreateNft.vue:378]: What's for?

**Review by roiLeo — APPROVED** [review]

File: components/create/CreateNft.vue (hunk 2 above)
Comment (prury) [reply components/create/CreateNft.vue]: changed to the proper generic name

**Review by preschian — APPROVED** [review]
already pointed by @roiLeo 👍🏻

File: components/create/CreateNft.vue (hunk 3 above)
Comment (daiagi) [reply components/create/CreateNft.vue:378]: Chain symbol is the fees asset chosen by user.
This is only relevant on Basilisk chain

Can be deleted when we deprecate bsx

File: components/create/CreateNft.vue (hunk 1 above)
Comment (daiagi) [reply components/create/CreateNft.vue]: done

File: components/create/CreateNft.vue (hunk 3 above)
Comment (roiLeo) [reply components/create/CreateNft.vue:378]: hmmmm

**Comment by yangwao** [comment]
Thanks!
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 5.26 USD/DOT ~ 9.506 $DOT
🧗 [1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8](https://polkadot.subscan.io/account/1cAsKZYNRb8dkSCpn4eVkEn6ycNZTGoDo5dGDgB8J1UUWK8)
🔗 [0x79639a6c0d8752f928c21b761f019a980b8fe83ce798525a250f0bdfc3f294fe](https://polkadot.subscan.io/extrinsic/0x79639a6c0d8752f928c21b761f019a980b8fe83ce798525a250f0bdfc3f294fe)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**